gdk4/src/subclass/content_provider.rs

```diff
@@ -0,0 +1,248 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{Clipboard, ContentFormats, ContentProvider};
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, Object, Value};
+
+pub trait ContentProviderImpl: ContentProviderImplExt + ObjectImpl {
+    fn content_changed(&self, provider: &Self::Type) {
+        self.parent_content_changed(provider)
+    }
+
+    fn attach_clipboard(&self, provider: &Self::Type, clipboard: &Clipboard) {
+        self.parent_attach_clipboard(provider, clipboard)
+    }
+
+    fn detach_clipboard(&self, provider: &Self::Type, clipboard: &Clipboard) {
+        self.parent_detach_clipboard(provider, clipboard)
+    }
+
+    fn ref_formats(&self, provider: &Self::Type) -> ContentFormats {
```

Comment:
Why is this called `ref` and not just `get`?

Reply:
Good question, I don't think there is any strong reason judging by the users in gdk/gtk.

Reply:
With the new naming, these would just become `fn formats()` and `fn parent_formats()` btw.

Reply:
ok